### PR TITLE
add possibility to override status handling

### PR DESCRIPTION
**Pull request by senid231**
add status_handlers to JsonApiClient::Resource.connection_options

### CODE REVIEW

File: lib/json_api_client/middleware/status.rb [hunk 1]
```diff
@@ -15,9 +20,24 @@ def call(environment)
...
       end
 
-      protected
+      private
+
+      def custom_handlers
+        Array.wrap @options[:custom_handlers]
+      end
+
+      def handled_by_custom?(code, env)
```
Comment (senid231) [inline lib/json_api_client/middleware/status.rb]: not sure about this name

**Comment by gaorlov** [comment]
I love this idea! Thanks a ton for taking the time. I have some hesitation around the `throw` interface, however. I see that `custom_handlers` is intended to act as an array; how would I define more than one? Would it look something like
```
MyApi::Base.connection_options[:status_handlers] = -> ( code, _env ) do
  throw(:handled) if code == 400
end,
-> ( code, _env ) do 
  # handle other code
end

# or

MyApi::Base.connection_options[:status_handlers] = handle_400, handle_404

# where handle_400 and handle_404 are lambdas
```

But that also means that if there's more than one handler whichever one gets hit first will short circuit the entire sequence, right? 

Also: I have never seen `throw` as a sign of success; is this common practice?

I don't know if any of this feels like an issue to you, 

```
class 400Handler


MyApi::Base.connection_options[:status_handlers] = { 400 => 40andler
                                                                                       404 => 404_handler,

**Comment by gaorlov** [comment]
sorry about the close/reopen. i am bad at laptop mouse use : (

**Comment by senid231** [comment]
@gaorlov good suggestions. Thank you for the review.

I've reworked the PR. Please, take a look.

**Pull request merged**